src/core/fonts.js

```diff
@@ -3351,9 +3352,119 @@ var Font = (function FontClosure() {
       return (charsCache[charsCacheKey] = glyphs);
     },
 
+    getGlyphsAndCharPosition: function Font_getGlyphsAndCharPosition(chars) {
```

Comment:
Nit: Here and elsewhere below, when adding *new* code you can simply format methods as `getGlyphsAndCharPosition(chars) {` instead.

Reply:
Also, this is an *almost* verbatim copy of the existing `charsToGlyphs` method, which feels quite unfortunate!
Furthermore, this ignores the existing caching that's used in that method, which can lead to quite a bit of unnecessary re-parsing here.

Given how this is used, with only one call-site as far as I can tell, could you not simply call both the *existing* `charsToGlyphs` and then the *new* `getCharPosition` methods in place of this "combine" one!?
(Considering the current lack of caching in the new methods, I really don't see that being much slower overall.)

Reply:
You're right: I removed that stuff.